controllers/ingressnodefirewallconfig_controller.go

```diff
@@ -0,0 +1,137 @@
+	IngressNodeFirewallManifestPath = "./bindata/manifests/daemon"
+)
+
+var ManifestPath = IngressNodeFirewallManifestPath
+
+// IngressNodeFirewallConfigReconciler reconciles a IngressNodeFirewallConfig object
+type IngressNodeFirewallConfigReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+	Log    logr.Logger
+}
+
+//+kubebuilder:rbac:groups=ingress-nodefw.ingress-nodefw,resources=ingressnodefirewallconfigs,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=ingress-nodefw.ingress-nodefw,resources=ingressnodefirewallconfigs/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=ingress-nodefw.ingress-nodefw,resources=ingressnodefirewallconfigs/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the IngressNodeFirewallConfig object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.12.1/pkg/reconcile
+func (r *IngressNodeFirewallConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	_ = context.Background()
+	logger := r.Log.WithValues("ingress node firewall config", req.NamespacedName)
+
+	instance := &ingressnodefwv1alpha1.IngressNodeFirewallConfig{}
+	err := r.Get(context.TODO(), req.NamespacedName, instance)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
+	}
+
+	if req.Name != defaultIngressNodeFirwallCrName {
+		logger.Error(err, "Invalid IngressNode firewall config resource name", "name", req.Name)
+		return ctrl.Result{}, nil // Return success to avoid requeue
+	}
+
+	if err := r.syncIngressNodeFwConfigResources(instance); err != nil {
+		return ctrl.Result{}, nil
+	}
+	return ctrl.Result{}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *IngressNodeFirewallConfigReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&ingressnodefwv1alpha1.IngressNodeFirewallConfig{}).
+		Complete(r)
+}
+
+func (r *IngressNodeFirewallConfigReconciler) syncIngressNodeFwConfigResources(config *ingressnodefwv1alpha1.IngressNodeFirewallConfig) error {
+	logger := r.Log.WithName("syncIngressNodeFirewallConfigResources")
+	logger.Info("Start")
+	data := render.MakeRenderData()
+
+	data.Data["Image"] = os.Getenv("DAEMONSET_IMAGE")
```

Comment:
Instead of taking DAEMONSET_IMAGE from env, shouldn't it come from config.Spec.Image?

Reply:
we aren't using the spec and we don't need it we will rely on envvar setting I will remove it from the CR